### PR TITLE
feat: gitlab-generic and gitlab-golang use installer

### PR DESCRIPTION
**Pull request by steinliber**
Signed-off-by: Meng JiaFeng <jiafeng.meng@merico.dev>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

## Description
- gitlab ci plugin use ci installer
- change gitlab ci options

## Related Issues
#952

### CODE REVIEW

File: internal/pkg/plugininstaller/ci/installer.go [hunk 1]
```diff
@@ -55,5 +55,29 @@ func DeleteCIFiles(options plugininstaller.RawOptions) error {
 		CommitMsg:  deleteCommitMsg,
 	}
 	return gitClient.DeleteFiles(commitInfo)
+}
+
+func PushCIContent(options plugininstaller.RawOptions) error {
```
Comment (aFlyBird0) [inline internal/pkg/plugininstaller/ci/installer.go:60]: This function is as same as `PushCIFiles`.

image

File: internal/pkg/plugininstaller/ci/installer.go [hunk 2]
```diff
@@ -55,5 +55,29 @@ func DeleteCIFiles(options plugininstaller.RawOptions) error {
...
+	if err != nil {
+		return err
+	}
+	//4. push ci files to git repo
+	_, err = gitClient.PushLocalFileToRepo(&git.CommitInfo{
+		CommitMsg:    defaultCommitMsg,
+		CommitBranch: defaultBranch,
+		GitFileMap:   gitMap,
+	}, true)
+	return err
```
Comment (aFlyBird0) [inline internal/pkg/plugininstaller/ci/installer.go:82]: redundant blank line

**Review by aFlyBird0 — CHANGES_REQUESTED** [review]

File: internal/pkg/show/config/plugins/gitlabci-generic.yaml [hunk 3]
```diff
@@ -7,13 +7,13 @@ tools:
...
+      projectRepo:
+        owner: YOUR_GITLAB_USERNAME
+        repo: YOUR_GITLAB_REPO_NAME
+        branch: YOUR_GITLAB_REPO_BRANCH
+        base_url: YOUR_GITLAB_HOST_ADDRESS
+      ci:
+        remoteURL : https://someplace.com/to/download/your/template
+        vars:
+          key1: value1
+          key2: value2
```
Comment (aFlyBird0) [inline internal/pkg/show/config/plugins/gitlabci-generic.yaml:19]: image

File: internal/pkg/show/config/plugins/gitlabci-golang.yaml [hunk 4]
```diff
@@ -7,7 +7,8 @@ tools:
...
     options:
-      # owner/repo; "path with namespace" is only GitLab API's way of saying the same thing; please change the values below.
-      pathWithNamespace: YOUR_GITLAB_USERNAME/YOUR_GITLAB_REPO_NAME
-      # main branch of the repo (to which branch the plugin will submit the workflows)
-      branch: main
+      projectRepo:
+        owner: YOUR_GITLAB_USERNAME
+        repo: YOUR_GITLAB_REPO_NAME
+        branch: YOUR_GITLAB_REPO_BRANCH
+        base_url: YOUR_GITLAB_HOST_ADDRESS
```
Comment (aFlyBird0) [inline internal/pkg/show/config/plugins/gitlabci-golang.yaml:14]: No new line at end of file

**Review by aFlyBird0 — APPROVED** [review]

**Review by daniel-hutao — APPROVED** [review]

**Pull request merged**